packages/devtools_app/test/flutter/timeline_screen_test.dart

```diff
@@ -41,62 +77,147 @@ void main() {
+
+    testWidgets('record', (WidgetTester tester) async {
+      await buildTimelineScreen(tester, TimelineMode.full);
+
+      // Verify initial state.
+      expect(
+        find.byKey(TimelineScreen.recordingInstructionsKey),
+        findsOneWidget,
+      );
+      expect(
+        find.byKey(TimelineScreen.recordingStatusKey),
+        findsNothing,
+      );
+
+      // Start recording.
+      await tester.tap(find.byKey(TimelineScreen.recordButtonKey));
+      await tester.pump();
+      expect(
+        find.byKey(TimelineScreen.recordingInstructionsKey),
+        findsNothing,
+      );
+      expect(
+        find.byKey(TimelineScreen.recordingStatusKey),
+        findsOneWidget,
+      );
+
+      // Stop recording.
+      await tester.tap(find.byKey(TimelineScreen.stopRecordingButtonKey));
+      await tester.pump();
+      expect(
+        find.byKey(TimelineScreen.recordingInstructionsKey),
+        findsNothing,
+      );
+      expect(
+        find.byKey(TimelineScreen.recordingStatusKey),
+        findsNothing,
+      );
+      expect(
+        find.byKey(TimelineScreen.emptyTimelineRecordingKey),
+        findsOneWidget,
+      );
+    });
+
+    testWidgets('clear', (WidgetTester tester) async {
+      // Clear the frame-based timeline.
+      await buildTimelineWithSelectedFrame(tester);
+      expect(find.byType(FlutterFramesChart), findsOneWidget);
+      expect(find.byKey(TimelineScreen.flameChartSectionKey), findsOneWidget);
+      expect(find.byType(TimelineFlameChart), findsOneWidget);
+      expect(find.byKey(TimelineScreen.recordingInstructionsKey), findsNothing);
+      expect(find.byType(EventDetails), findsOneWidget);
+
+      await tester.tap(find.byKey(TimelineScreen.clearButtonKey));
+      await tester.pump();
+      expect(find.byType(FlutterFramesChart), findsOneWidget);
+      expect(find.byKey(TimelineScreen.flameChartSectionKey), findsNothing);
+      expect(find.byType(EventDetails), findsNothing);
+
+      // Clear the full timeline.
+      await buildTimelineScreen(tester, TimelineMode.full);
+      await tester.tap(find.byKey(TimelineScreen.recordButtonKey));
+      await tester.pump();
+      await tester.tap(find.byKey(TimelineScreen.stopRecordingButtonKey));
+      await tester.pump();
+      expect(
+        find.byKey(TimelineScreen.recordingInstructionsKey),
+        findsNothing,
+      );
+      expect(
+        find.byKey(TimelineScreen.recordingStatusKey),
+        findsNothing,
+      );
+      expect(
+        find.byKey(TimelineScreen.emptyTimelineRecordingKey),
+        findsOneWidget,
+      );
+
+      await tester.tap(find.byKey(TimelineScreen.clearButtonKey));
+      await tester.pump();
+      expect(
+        find.byKey(TimelineScreen.recordingInstructionsKey),
+        findsOneWidget,
+      );
+      expect(
+        find.byKey(TimelineScreen.recordingStatusKey),
+        findsNothing,
+      );
+      expect(
+        find.byKey(TimelineScreen.emptyTimelineRecordingKey),
```

Comment:
If there's any logic that these changes are supposed to show (like a method call on a mock controller), check those too.